### PR TITLE
[CFM-1313] Reactivate user link + notification.

### PR DESCRIPTION
**Pull request by anvmn**
https://issuetracker.amplexor.com/jira/browse/CFM-1313

selection_096

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_admin/c4m_user_admin.pages.inc [hunk 1]
```diff
@@ -28,6 +28,34 @@ function c4m_user_admin_user_activate_form($form, &$form_state, $account) {
...
+ * @see c4m_user_admin_user_reactivate_form_submit()
+ *
+ * @ingroup forms
+ */
+function c4m_user_admin_user_reactivate_form($form, &$form_state, $account) {
+  $form_state['user'] = $account;
+
+  $account = entity_metadata_wrapper('user', $account);
+  $question = t('Are you sure you want to reactivate the account of @first_name @last_name?', array(
+    '@first_name' => $account->c4m_first_name->value(),
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_admin/c4m_user_admin.pages.inc]: use `c4m_user_name()`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_admin/include/views_handler_field_user_link_reactivate.inc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  /**
+   * @inheritdoc
+   */
+  public function render_link($data, $values) {
+    $uid = $values->{$this->aliases['uid']};
+
+    $account = user_load($uid);
+    $account = entity_metadata_wrapper('user', $account);
+
+    // No need to check c4m_is_deleted, because it's tightly coupled with status.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_admin/include/views_handler_field_user_link_reactivate.inc]: please elaborate about the __last login__

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_admin/include/views_handler_field_user_link_activate.inc [hunk 3]
```diff
@@ -22,7 +22,8 @@ class views_handler_field_user_link_activate extends views_handler_field_user_li
     $account = user_load($uid);
     $account = entity_metadata_wrapper('user', $account);
 
-    if (!$account->status->value() && !$account->c4m_is_spammer->value() && !$account->c4m_is_deleted->value() && user_access('administer users')) {
+    // No need to check c4m_is_deleted, because it's tightly coupled with status.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_admin/include/views_handler_field_user_link_activate.inc]: please elaborate about the __last login__

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_admin/include/views_handler_field_user_link_resend_activation_email.inc [hunk 4]
```diff
@@ -22,7 +22,8 @@ class views_handler_field_user_link_resend_activation_email extends views_handle
     $account = user_load($uid);
     $account = entity_metadata_wrapper('user', $account);
 
-    if (!$account->status->value() && !$account->c4m_is_spammer->value() && !$account->c4m_is_deleted->value() && user_access('administer users')) {
+    // No need to check c4m_is_deleted, because it's tightly coupled with status.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_admin/include/views_handler_field_user_link_resend_activation_email.inc]: please elaborate about the __last login__

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_admin/c4m_user_admin.pages.inc [hunk 5]
```diff
@@ -39,6 +67,22 @@ function c4m_user_admin_user_activate_form_submit($form, &$form_state) {
...
+ * @see c4m_user_admin_user_activate_form()
+ */
+function c4m_user_admin_user_reactivate_form_submit($form, &$form_state) {
+  c4m_user_admin_user_activate_form_submit($form, $form_state);
+
+  if ($form_state['values']['c4m_notify_user']){
+    // Send reactivate notification.
+    $account = $form_state['user'];
+    $params['account'] = $account;
+    drupal_mail('user_registrationpassword', 'register', $account->mail, user_preferred_language($account), $params);
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_admin/c4m_user_admin.pages.inc]: why not using `_c4m_user_send_mail()` here ?

Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_admin/c4m_user_admin.pages.inc]: @ordavidil 
We send here a mail using old mailing system.
Mail subject and body are located within txt files at /capacity4more/modules/c4m/mail/c4m_mail/messages.
_c4m_user_send_mail() expects to receive subject and body as params.
It's also defined at CFM 1200, which is not merged yet.

**Pull request merged**